pkg/monitor/api.go

```diff
@@ -58,6 +58,56 @@ func startAPIMonitoring(ctx context.Context, m *Monitor, clusterConfig *rest.Con
 		return err
 	}
 
+	// watch individual kube-apiservers and specifically report their healthz
+	kubeClient, err := clientcorev1.NewForConfig(clusterConfig)
+	if err != nil {
+		return err
+	}
+	allKubeAPIServerPods, err := kubeClient.Pods("openshift-kube-apiserver").List(metav1.ListOptions{})
+	if err == nil {
+		for _, pod := range allKubeAPIServerPods.Items {
+			// this a reasonable signal that we're an install pod for this ns
+			if !pod.Spec.HostNetwork {
+				continue
+			}
+			perKubeAPIServerPollingClientConfig := rest.CopyConfig(clusterConfig)
+			perKubeAPIServerPollingClientConfig.Timeout = 3 * time.Second
+			perKubeAPIServerPollingClientConfig.ServerName = "kubernetes.default.svc"
+			perKubeAPIServerPollingClientConfig.Host = pod.Status.HostIP + ":6443"
```

Comment:
This isn't going to work if Erics changes land to block accessing pods directly.

Reply:
This is the only real blocker here - is there another way to do this without relying on public IP addresses on the masters?

Reply:
> This isn't going to work if Erics changes land to block accessing pods directly.

Not that I know.  We run outside the cluster and we're trying to directly get at a hostNetwork pod.